### ISSUE TITLE
Overlay Scrollbars Freezes Browser

### ISSUE THREAD

**Issue by FunkMonkey33** [KingSora/OverlayScrollbars#152]
Using Google Chrome.  Loading OverlayScrollbars via CDN.  It is in an Angular application, but I'm using the plain, non-wrapped version.

I'm running it like this:

In NgOnInit:

    $(document).ready(() => {
      OverlayScrollbars(document.querySelectorAll('#MainContentSection'), {});
    });

The HTML template looks like this:

<div id="MainContentSection" style="max-height: 500px; overflow: hidden;" >
    ...
</div>

The first time the page loads, this works just fine.  If I go to a different view, however, the browser freezes.  Could this be a memory leak?

Thanks.

Aaron

**Comment by KingSora** [comment]
Good day!

Thanks for your report, could you create a small example on [Stackblitz](https://stackblitz.com/) or [CodeSandbox](https://codesandbox.io/)? It would make fixing the problem much easier for me!

Thanks

**Comment by KingSora** [comment]
@FunkMonkey33 any update here?

**Comment by Teranode** [comment]
yea this is getting worse now

**Comment by FunkMonkey33** [comment]
Sorry, Rene,

We let go of using this component and went with another solution.  I won't
be able to return to this any time soon due to workload.

Aaron

**Comment by KingSora** [comment]
@Teranode in this case, I would ask you to create a small example on [Stackblitz](https://stackblitz.com/) or [CodeSandbox](https://codesandbox.io/).

**Comment by KingSora** [comment]
@Teranode @FunkMonkey33 
After the author of #168 had a similar issue and could provide a example, I've figured out how to fix it. The issue is related to Angulars `zone.js` in connection with `MutationObserver`. The fix for the `overlayscrollbars-ngx` package will be released with the next release `v0.3.0`.

In case you want to initialize `OverlayScrollbars` by yourself, do it like this:

```ts
import { ElementRef, AfterViewInit, NgZone } from '@angular/core';
import OverlayScrollbars from 'overlayscrollbars';

export class OverlayScrollbarsComponent implements AfterViewInit {
    private _osInstance: OverlayScrollbars | null = null;
    private _osTargetRef: ElementRef;
    
    constructor(_osTargetRef: ElementRef, private ngZone: NgZone) {
        this._osTargetRef = _osTargetRef;
    }

    ngAfterViewInit() {
        this.ngZone.runOutsideAngular(() => {
            this._osInstance = OverlayScrollbars(this.osTarget(), this._options || {}, this._extensions);
        });
    }
}
```

This code is just a snipped and won't work completely. The important part is the `ngAfterViewInit` function where `OverlayScrollbars` is initialized in the `this.ngZone.runOutsideAngular` function. If you wanna see the complete code you can find it [here](https://github.com/KingSora/OverlayScrollbars/blob/master/packages/overlayscrollbars-ngx/src/overlayscrollbars.component.ts).
